File: Leetcode/JS/Leetcode71-简化路径.js
var simplifyPath = function(path) {
    let res = '',name = '';
    if(path[path.length - 1] != '/') path += '/';
    for(const ch of path){
        if(ch != '/') name += ch;
        else{
            if(name == '..'){
                while(res.length && res[res.length - 1] != '/') res = res.slice(0,res.length - 1);
                if(res.length) res = res.slice(0,res.length - 1);
            }else if(name != '.' && name != ''){
                res += '/' + name;
            }
            name = '';
        }
    }
    if(res.length == 0) res = '/';
    return res;
};

File: Leetcode/PY/Leetcode71-简化路径.py
class Solution:
    def simplifyPath(self, s: str) -> str:
        if s[-1] != '/': s += '/'
        res,name = '',''
        for ch in s:
            if ch != '/': name += ch
            else:
                if name == '..':
                    while res and res[-1] != '/': res = res[:-1]
                    if res: res = res[:-1]
                elif name != '.' and name != '':
                    res += '/' + name
                name = ''
        if not res: res = '/'
        return res